pennylane/ops/qubit/matrix_ops.py

```diff
@@ -400,17 +400,29 @@ class BlockEncode(Operation):
     num_wires = AnyWires
     """int: Number of wires that the operator acts on."""
 
+    ndim_params = (2,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
     def __init__(self, A, wires, do_queue=True, id=None):
-        A = qml.math.atleast_2d(A)
+        # A = qml.math.atleast_2d(A)
+        if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
```

Comment:
```suggestion
        if qml.math.shape(A) in [(), (1,)]:
```
Optional suggestion (prevents computing shape twice).

Reply:
:brain: